File: backend/app/schemas/ai.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class AiExplainRequest(BaseModel):
    vehicle_id: UUID
    question: str
    question_type: str = "general"


class AiExplanationOut(BaseModel):
    id: UUID
    vehicle_id: UUID | None
    asked_at: datetime
    user_question: str
    answer_markdown: str | None
    confidence: str | None
    provider: str | None
    model: str | None
    error: str | None

    model_config = {"from_attributes": True}

File: frontend/src/types/ai.ts
export interface AiExplanation {
  id: string;
  vehicle_id: string | null;
  asked_at: string;
  user_question: string;
  answer_markdown: string | null;
  confidence: string | null;
  provider: string | null;
  model: string | null;
  error: string | null;
}

export interface AiExplainRequest {
  vehicle_id: string;
  question: string;
  question_type: string;
}

export type QuestionType =
  | "general"
  | "vampire_drain"
  | "battery_health"
  | "charging_efficiency"
  | "costs";
